lightning/src/ln/channelmanager.rs

```diff
@@ -2397,6 +2404,55 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		self.process_background_events();
 	}
 
+	fn update_channel_fee(&self, short_to_id: &mut HashMap<u64, [u8; 32]>, pending_msg_events: &mut Vec<events::MessageSendEvent>, chan_id: &[u8; 32], chan: &mut Channel<Signer>, new_feerate: u32) -> (bool, NotifyOption, Result<(), MsgHandleErrInternal>) {
+		let mut retain_channel = true;
+		let mut ret_err = Ok(());
+		let mut should_persist = NotifyOption::SkipPersist;
+		if chan.is_outbound() {
+			if chan.get_feerate() < new_feerate || chan.get_feerate() > new_feerate * 2 {
+				log_trace!(self.logger, "Channel {} qualifies for a feerate change from {} to {}. Checking if its live ({}).",
+					log_bytes!(chan_id[..]), chan.get_feerate(), new_feerate, chan.is_live());
+				if chan.is_live() {
+					should_persist = NotifyOption::DoPersist;
+					let res = match chan.send_update_fee_and_commit(new_feerate, &self.logger) {
```

Comment:
Do we check that the proposed update_fee (or received!) is at least equal or superior to mempool min relay fee ? The check enforced here : https://github.com/bitcoin/bitcoin/blob/531c2b7c04898f5a2097f44e8c12bfb2f53aaf9b/src/validation.cpp#L519

I don't see it in `Channel::check_remote_fee` or `send_update_fee`. We have a worst-case check on the receiver-side on `ConfirmationTarget::Background`, but not on the sender. And we might also have it as a belt-and-suspender in case of buggy/compromised fee-estimators ?

Reply:
Hmm, true, I think we need to apply that everywhere we call `get_est_sat_per_1000_weight`, though, not just here. Maybe we can open an issue and add assertions everywhere we call it later?

Reply:
Tracked : https://github.com/rust-bitcoin/rust-lightning/issues/1016